.github/workflows/update_protos.yml

```diff
@@ -46,12 +46,14 @@ jobs:
           BOOST_ROOT: ${{ github.workspace }}/boost/boost
       - name: cleanup
         run: rm -rf boost/ && rm -rf vcpkg/
+      - name: Update proto tag
```

Comment:
Yeah so I had a little bash script for getting the version tag based on the commit in the `buf.lock` file but I ended up going against that approach because it felt a little fragile (required internet access, relied on the output of `buf beta registry label list` maintaining a consistent format). I didn't see any other good way to get the tag (it doesn't seem to exist in the sources anywhere).

> Also, how does it work above that doing the `build --target update-static-protos` is building against the version of the protos that we are then interpolating?

Sorry, I'm not sure I understand what you're asking here exactly. Can you rephrase?

Reply:
> > Also, how does it work above that doing the `build --target update-static-protos` is building against the version of the protos that we are then interpolating?
> 
> Sorry, I'm not sure I understand what you're asking here exactly. Can you rephrase?

Let me try again. Basically, I'd like to understand better what the workflow does. How does it end up building the protos against a specific version of the API repo? Can *that* process be adjusted such that the API repo version tag can be programatically read out of the static protos in the tree, rather than needing to edit a constant in the CMakeLists.

Reply:
Ah, the workflow builds the latest version and runs whenever the protos are updated, then updates the commit in `buf.lock` to be whatever the latest commit is. It always builds against main. It gets the tag info from the `api` repo's `tag_and_release_protos` workflow. 

I'm not aware of anywhere currently where the tag exists in the tree such that we could get it without storing a constant somewhere. We _could_ write it to a separate file, but then we're still editing a constant somewhere and it complicates the process of building out the `version_metadata.cpp` file since we'd then be modifying it twice from two different sources and have to preserve both changes, so we'd probably need to have an intermediary file in between `version_metadata.cpp.in` and `version_metadata.cpp`. So I'm not sure if that's a great approach.

Alternatively, we could use the `buf` command line tool to get tag info and then some grep/sed magic to get the tag, but again I think this is pretty unreliable and probably not worth it.

Reply:
@stuqdog and I talked this over. Writing the version to a file under `src/viam/api` will simplify the script and also not give the appearance that this value is actually "always" true. It isn't true, for instance, if you are building against local protos, in which case the API version should be "unknown" or something more like `git describe` output.

Reply:
Do we want to always build against main? I feel like we should only build against the version tag that was provided. I think every time someone builds a proto, it should be the same, predictable output regardless of what's going on in API world. Only when the tag is changed (either manually or via this workflow) should the compiled output change.

Reply:
In general, every time someone builds a proto it _is_ the same, predictable output. The one exception that we're worried about here is when someone explicitly opts in to building against a local API definition (if, e.g., they wanted to test out API changes within the C++ SDK). This is explicitly opt-in and unless someone is specifically looking for it they'll never encounter or have to think about it. If we no longer want to offer that as an option we can certainly discuss, but I'd argue that such a change is out of scope for this PR.